allensdk/internal/api/__init__.py

```diff
@@ -66,3 +73,35 @@ def select_one(self, query):
         if len(data) == 1:
             return data[0]
         return {}
+
+
+def db_connection_creator(default_credentials: dict,
```

Comment:
I know that you're trying to minimize repeated code, but this definitely made it more confusing for me.

At least I'd ask to update the docstring here. Both what you're calling "default_credentials" and "credentials" are optional -- exactly one is required. 

`credentials` is an explicit mapping from the kwarg for the psycopg connection to the value that needs to be injected. e.g. {"dbname": "mydb"}

what you're calling `default_credentials` is a mapping from an environment variable (or in the future some other arbitrary key in a supported `CredentialProvider`) to the kwarg in the psycopg2 connection. e.g. {"dbname": "MYDB_NAME"}, and then you'd have `export MYDB_NAME="mydb_name"` somewhere.

I'm just not sure that's very clear here.